File: bloaters/long-method/extract-method/PHP/problem.php
<?php

namespace PHP;

class Problem
{
    public $name;
    public $amount;

    function __construct()
    {
    }

    function printResult(): void
    {
        $this->printHeader();

        // Print details
        echo "Name: {$this->name}\n";
        echo "Amount by Length of Name: {$this->getAmountByLengthOfName()}\n";
    }

    private function printHeader(): void
    {
        echo "The result is:\n";
    }

    private function getAmountByLengthOfName()
    {
        return $this->amount * strlen($this->name);
    }
}

File: bloaters/long-method/extract-method/C#/Problem.cs
using System;

namespace CSharp
{
    public class Problem
    {
        public string Name { get; set; }
        public double Amount { get; set; }

        public Problem()
        {
        }

        public void PrintResult()
        {
            PrintHeader();

            // Print Details
            Console.WriteLine($"Name: {Name}");
            Console.WriteLine($"Amount by Length of Name: {GetAmountByLengthOfName()}");
        }

        private void PrintHeader()
        {
            Console.WriteLine("The result is:");
        }

        private double GetAmountByLengthOfName()
        {
            return Amount * Name.Length;
        }
    }
}
